ocaml/xapi/xapi_vbd.ml

```diff
@@ -47,10 +77,23 @@ let plug ~__context ~self =
     Storage_access.attach_and_activate ~__context ~vbd:self ~domid
       (fun attach_info ->
          let params = attach_info.Storage_interface.params in
-         let prefix = "/dev/" in
-         let prefix_len = String.length prefix in
-         let path = String.sub params prefix_len (String.length params - prefix_len) in
-         Db.VBD.set_device ~__context ~self ~value:path;
+         let device_path =
+           let nbd_prefix = "hack|nbd:unix:" in
```

Comment:
:disappointed:

Reply:
We will come up with a proper design, but for now, the strategy is to make sure the new code is isolated, and this `"hack|..."` method is hopefully good enough for that.